alf/environments/torch_wrappers.py

```diff
@@ -118,7 +118,10 @@ def _step(self, action):
 
         self._num_steps += 1
         if self._num_steps >= self._duration:
-            time_step = time_step._replace(step_type=StepType.LAST)
+            LAST = StepType.LAST
+            if time_step.step_type.is_cuda:
+                LAST = LAST.cuda()
+            time_step = time_step._replace(step_type=LAST)
```

Comment:
I think the more principled way (and less error-prone in the future) is to make sure that when you init step types in `data_structures.py` the tensors are on the right device?

Reply:
Just check if cuda is available, if yes, then manually specify there devices

Reply:
Probably due to StepType.LAST being a class constant and executed during import of the class, while set_default_device only happens when we start executing alf.set_default_device("cuda") in train.py?  So torch defaults the tensor to be CPU?

Reply:
We can change them to numpy int: np.int32(0), np.int32(1), np.int32(2).

Reply:
Updated, but need some luck to pass the ddpg test.